src/components/tx/SignOrExecuteForm/SignOrExecuteForm.test.tsx

```diff
@@ -0,0 +1,209 @@
+import { act, fireEvent, render } from '@/tests/test-utils'
+import SignOrExecuteForm from '@/components/tx/SignOrExecuteForm/index'
+import { SafeSignature, SafeTransaction } from '@gnosis.pm/safe-core-sdk-types'
+import * as useSafeInfoHook from '@/hooks/useSafeInfo'
+import * as useGasLimitHook from '@/hooks/useGasLimit'
+import * as txSender from '@/services/tx/txSender'
+import * as wallet from '@/hooks/wallets/useWallet'
+import { SafeInfo, TransactionDetails } from '@gnosis.pm/safe-react-gateway-sdk'
+import { waitFor } from '@testing-library/react'
+import { ConnectedWallet } from '@/services/onboard'
+import * as safeCoreSDK from '@/hooks/coreSDK/safeCoreSDK'
+import Safe from '@gnosis.pm/safe-core-sdk'
+
+const createSafeTx = (): SafeTransaction => {
+  return {
+    data: {
+      to: '0x0000000000000000000000000000000000000000',
+      value: '0x0',
+      data: '0x',
+      operation: 0,
+      nonce: 100,
+    },
+    signatures: new Map([]),
+    addSignature: function (sig: SafeSignature): void {
+      this.signatures.set(sig.signer, sig)
+    },
+    encodedSignatures: function (): string {
+      return Array.from(this.signatures)
+        .map(([, sig]) => {
+          return [sig.signer, sig.data].join(' = ')
+        })
+        .join('; ')
+    },
+  } as SafeTransaction
+}
+
+describe('SignOrExecuteForm', () => {
+  let mockSDK
+
+  beforeEach(() => {
+    jest.resetAllMocks()
+
+    mockSDK = {
+      isModuleEnabled: jest.fn(() => false),
+      createTransaction: jest.fn(() => 'asd'),
+      getTransactionHash: jest.fn(() => '0x10'),
+    } as unknown as Safe
+
+    jest.spyOn(safeCoreSDK, 'getSafeSDK').mockReturnValue(mockSDK)
+    jest.spyOn(useSafeInfoHook, 'default').mockImplementation(() => ({
+      safe: {
+        nonce: 100,
+      } as SafeInfo,
+      safeAddress: '0x123',
```

Comment:
Currently its only there so that the type check passes but its not used by the test. I think in the future we could consider adding factory methods for the bigger types we have so mocking is a bit easier.